src/sdi-user-session-helper.c

```diff
@@ -111,7 +119,40 @@ static void new_session(Login1Manager *manager, const gchar *session_id,
   }
 }
 
-void sdi_wait_for_graphical_session(void) {
+static void dbus_timeout_cb(GMainLoop *loop) {
+  dbus_timeout_id = 0;
+  g_main_loop_quit(loop);
+}
+
+static gboolean check_dbus_names_cb(GMainLoop *loop) {
+  g_auto(GStrv) names = NULL;
+  GError *error = NULL;
+  dbus_server__call_list_names_sync(dbus_server, &names, NULL, &error);
```

Comment:
This can be quite slow, it would be nicer to have it async, although it's also true we're still blocking on this... :thinking:

Reply:
That's the point: we are already blocking here, so let's not overcomplicate things.